packages/blitz-dom/src/events/driver.rs

```diff
@@ -55,8 +55,20 @@ impl<'doc, Handler: EventHandler> EventDriver<'doc, Handler> {
             UiEvent::MouseMove(event) => {
                 let dom_x = event.x + viewport_scroll.x as f32 / zoom;
                 let dom_y = event.y + viewport_scroll.y as f32 / zoom;
-                self.doc_mut().set_hover_to(dom_x, dom_y);
+                let changed = self.doc_mut().set_hover_to(dom_x, dom_y);
+                if changed {
+                    if let Some(target) = hover_node_id {
+                        self.handle_dom_event(DomEvent::new(target, DomEventData::MouseOut(event.clone())));
+                        self.handle_dom_event(DomEvent::new(target, DomEventData::MouseLeave(event.clone())));
+                    }
+                }
                 hover_node_id = self.doc().hover_node_id;
+                if changed {
+                    if let Some(target) = hover_node_id {
+                        self.handle_dom_event(DomEvent::new(target, DomEventData::MouseOver(event.clone())));
+                        self.handle_dom_event(DomEvent::new(target, DomEventData::MouseEnter(event.clone())));
+                    }
+                }
```

Comment:
This looks correct for `mouseover`/`mouseout`. But I think for `mouseenter`/`mouseleave` you actually need to diff the "node stack" against the previous "node stack", and individually send the event to each node that has been entered or left.

See: https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseenter_event#usage_notes